src/Forms/Form.php

```diff
@@ -356,12 +356,22 @@ public function requireSudoMode(): void
             return;
         }
         // If sudo mode is not active, make the form readonly and add a sudo mode password field
+        // assuming that wasn't already there
         $this->makeReadonly();
-        $field = SudoModePasswordField::create();
-        // Manually call setForm() to the field as the field list is being updated after the
-        // form is created, which is when setForm() is normally being created
-        $field->setForm($this);
-        $this->Fields()->unshift($field);
+        $hasSudoModeField = false;
+        foreach ($this->Fields()->flattenFields() as $field) {
```

Comment:
Flatten fields results in calling `$field->setContainerFieldList()` in a new `FieldList`, which could cause weird behaviour (see https://github.com/silverstripe/silverstripe-framework/issues/10054). We should either:

1. Just assume it'll be in the root, because that's where we put it. If someone else is putting it nested somewhere then they can deal with the weirdness that causes themselves.
2. Use `recursiveWalk()` here directly (which `flattenFields()` uses internally)

Reply:
Have gone with assuming it's in the root